lib/reservation.rb

```diff
@@ -0,0 +1,47 @@
+require 'date'
+
+class Reservation
```

Comment:
I would suggest a method in reservation which either takes a `Reservation` or a date range and returns true if it overlaps with this reservation and false if it does not.  It might make the logic easier in Admin.